Makefile

```
@@ -1986,6 +1986,19 @@ $(asmgen_OBJECT): $(asmgen_SOURCE)
 	$(V_ASM)$(ASPP) $(OC_ASPPFLAGS) -o $@ $< || $(ASPP_ERROR)
 endif
 
+test_in_prefix_SOURCES = \
+  $(addprefix testsuite/tools/,test_in_prefix.mli test_in_prefix.ml)
+test_in_prefix_LIBRARIES = otherlibs/unix/unix compilerlibs/ocamlcommon
+
+testsuite/tools/test_in_prefi%: CAMLC = $(BEST_OCAMLC) $(STDLIBFLAGS)
+
+testsuite/tools/test_in_prefix$(EXE): OC_BYTECODE_LINKFLAGS += -custom
+
+testsuite/tools/test_in_prefi%: CAMLOPT = $(BEST_OCAMLOPT) $(STDLIBFLAGS)
```

```suggestion
testsuite/tools/test_in_prefix%: CAMLC = $(BEST_OCAMLC) $(STDLIBFLAGS)

testsuite/tools/test_in_prefix$(EXE): OC_BYTECODE_LINKFLAGS += -custom

testsuite/tools/test_in_prefix%: CAMLOPT = $(BEST_OCAMLOPT) $(STDLIBFLAGS)
```

Why drop the leading `O` in `OCAMLC` and `OCAMLOPT`? or is it always spelled this way?

---

> Why drop the leading `O` in `OCAMLC` and `OCAMLOPT`? or is it always spelled this way?

@shindere definitely wants to change this (the names are just legacy from Caml Light and Caml Special Light!)... I _think_ the idea was to wait until the `Makefile`s are all merged... renaming Makefile variables is of course a tricky process to get right (all the recursive calls, etc.)

---

Makefile patterns cannot match the empty string, which is why the `x` is omitted - this rule is intended to match _both_ `testsuite/tools/test_in_prefix` and `testsuite/tools/test_in_prefix.opt` and the pattern `testsuite/tools/test_in_prefix%` doesn't match the first one.

cf. the same trick in `stdlib/Makefile`:
https://github.com/dra27/ocaml/blob/f28bfbcb232b017460cb0ba7f9ec8cf1cf773811/stdlib/Makefile#L93-L94
where the pattern is `%-launch-info` (to match both `runtime-launch-info` and `target_runtime-launch-info`)

---

Interesting Makefile craziness... thanks!